src/listener/multi_listener.rs

```diff
@@ -0,0 +1,50 @@
+use crate::listener::{Listener, ToListener};
+use crate::utils::BoxFuture;
+use crate::Server;
+
+use async_std::io;
+use async_std::prelude::*;
+
+pub struct MultiListener<State>(Vec<Box<dyn Listener<State>>>);
+
+impl<State> std::fmt::Debug for MultiListener<State> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{:?}", self.0)
+    }
+}
+
+impl<State: 'static + Send + Sync> MultiListener<State> {
+    pub fn new() -> Self {
+        Self(vec![])
+    }
+
+    pub fn add<I: ToListener<State>>(&mut self, listener: I) {
+        self.0.push(listener.to_listener());
+    }
+
+    pub fn from_iter<TL: ToListener<State>>(vec: impl IntoIterator<Item = TL>) -> Self {
+        Self(
+            vec.into_iter()
+                .map(|listener| listener.to_listener())
+                .collect(),
+        )
+    }
+}
+
+impl<State: 'static + Send + Sync> Listener<State> for MultiListener<State> {
+    fn listen<'a>(&'a mut self, app: Server<State>) -> BoxFuture<'a, io::Result<()>> {
+        let mut fut: Option<BoxFuture<'a, io::Result<()>>> = None;
+
+        for listener in self.0.iter_mut() {
+            let app = app.clone();
+            let listened = listener.listen(app);
+            if let Some(f) = fut {
+                fut = Some(Box::pin(f.race(listened)));
```

Comment:
This is awkward and improvements are very welcome. I was looking for a way to race an iter of an arbitrary number of futures and this was what I came up with